### PR TITLE
change where the close button is in the dom in dialtone modal docs/examples

### PR DESCRIPTION
**Pull request by braddialpad**
update docs to portray correct tab order in modal

### CODE REVIEW

**Comment by dbecherdialpad** [comment]
Can we improve the Accessibility docs to mention this? I think [USWDS](https://designsystem.digital.gov/components/modal/) and [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) have good examples of focus management when the dialog is opened.

File: docs/components/modal.html [hunk 1]
```diff
@@ -94,12 +94,12 @@ <h3 class="dialtone-title__subsection">Best practices</h3>
 <aside class="d-modal" id="modal-base" tabindex="-1" role="dialog" aria-labelledby="modal-title" aria-describedby="modal-description" aria-hidden="true">
   <div class="d-modal__dialog" role="document">
     <h2 class="d-modal__header" id="modal-title">…</h2>
-    <button href="#" class="d-modal__close d-btn d-btn--circle d-btn--lg" aria-label="Close"><IconClose \></button>
     <p class="d-modal__content" id="modal-description">…</p>
     <footer class="d-modal__footer">
       <button class="d-btn" type="button">…</button>
       <button class="d-btn d-btn--primary" type="button">…</button>
     </footer>
+    <button href="#" class="d-modal__close d-btn d-btn--circle d-btn--lg" aria-label="Close"><IconClose \></button>
```
Comment (dbecherdialpad) [inline docs/components/modal.html]: nit:

```suggestion
    <button href="#" class="d-modal__close d-btn d-btn--circle d-btn--lg" aria-label="Close"><IconClose /></button>
```

**Comment by braddialpad** [comment]
Updated the modal accessibility docs with some additional details about tab focus and a link to MDN.

**Review by dbecherdialpad — APPROVED** [review]
lgtm!

**Pull request merged**